tests/components/honeywell/test_config_flow.py

```diff
@@ -33,28 +33,32 @@ async def test_show_authenticate_form(hass: HomeAssistant) -> None:
     assert result["step_id"] == "user"
 
 
-async def test_connection_error(hass: HomeAssistant) -> None:
+async def test_connection_error(hass: HomeAssistant, client: MagicMock) -> None:
+    """Test that an error message is shown on connection fail."""
+    client.login.side_effect = AIOSomecomfort.ConnectionError
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
+    )
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_auth_error(hass: HomeAssistant, client: MagicMock) -> None:
     """Test that an error message is shown on login fail."""
-    with patch(
-        "somecomfort.SomeComfort",
-        side_effect=somecomfort.AuthError,
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
-        )
-        assert result["errors"] == {"base": "invalid_auth"}
+    client.login.side_effect = AIOSomecomfort.AuthError
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
+    )
+    assert result["errors"] == {"base": "invalid_auth"}
 
 
 async def test_create_entry(hass: HomeAssistant) -> None:
     """Test that the config entry is created."""
-    with patch(
-        "somecomfort.SomeComfort",
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
-        )
-        assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
-        assert result["data"] == FAKE_CONFIG
+
+    result = await hass.config_entries.flow.async_init(
```

Comment:
Please patch the integration setup function on all create entry results to avoid setting up the whole integration. See the scaffold script test template code for example.